app/javascript/pages/Analytics/UtmLinks/Index.tsx

```diff
@@ -57,106 +36,144 @@ const utmLinkWithStats = (utmLink: SavedUtmLink, stats?: UtmLinkStats) => {
   return link;
 };
 
-const UtmLinkList = () => {
-  const navigation = useNavigation();
-  const navigate = useNavigate();
-  const revalidator = useRevalidator();
-  const { utm_links: utmLinks, pagination } = cast<{ utm_links: SavedUtmLink[]; pagination: PaginationProps }>(
-    useLoaderData(),
-  );
-  const [utmLinksStats, setUtmLinksStats] = React.useState<UtmLinksStats>({});
-  const utmLinksWithStats = utmLinks.map((utmLink) => utmLinkWithStats(utmLink, utmLinksStats[utmLink.id]));
+const extractSortParam = (rawParams: URLSearchParams): Sort<SortKey> | null => {
+  const column = rawParams.get("key");
+  switch (column) {
+    case "link":
+    case "date":
+    case "source":
+    case "medium":
+    case "campaign":
+    case "clicks":
+    case "sales_count":
+    case "revenue_cents":
+    case "conversion_rate":
+      return {
+        key: column,
+        direction: rawParams.get("direction") === "desc" ? "desc" : "asc",
+      };
+    default:
+      return null;
+  }
+};
```

Comment:
import { extractSortParam } from "$app/components/server-components/UtmLinksPage";

This component was inlined over here, so logic moved as well